### DISCUSSION TITLE
How to pass a variable between tests where the variable gets updated within each test.

### DISCUSSION THREAD

**Discussion by PatrickFlaherty** [pytest-dev/pytest#11583] [Q&A]
How to pass a variable between tests where the variable gets updated within each test within a module?
I see I can use a namespace within the pytest framework.
If for example I use pytest.my_variable, does this become a global variable such that I need to worry about using the same variable name in other tests running in parallel?

Thanks in advance for any input,
Patrick

**Comment by nicoddemus** [comment]
Hi @PatrickFlaherty,

Adding a variable to a module makes it global, this is how Python works and not related to pytest.

However it is recommended for tests to not depend on one another, so it is not recommended in general for tests to generate and pass information to one another.

**Reply by RonnyPfannschmidt** [reply]
To elaborate, tests altering the state of data from other tests easily creates a situation where one loses the ability to safely run tests in random order or just subsets/singular tests

This quickly can escalate/degrade into a fully blown loss of debugging/quick testing capabilities

**Comment by tmlnv** [comment]
As mentioned above, it is not advisable for tests to rely on each other. However, `pytest-dependency` can be a viable solution for managing shared data, especially when you have a sequence of tests that depend on the state generated by prior tests. By using `pytest-dependency`, you can utilize the `@pytest.mark.dependency` decorator to indicate these dependencies. To achieve the desired behavior, you can organize the tests within a test class and store data as a class attribute.

Here's a basic example to illustrate this, using a test class:
```python
import pytest


class TestSequence:
    shared_data = {}  # Class attribute to store shared data

    @pytest.mark.dependency(name="test_one")
    def test_one(self):
        TestSequence.shared_data['value'] = 'data from test_one'

    @pytest.mark.dependency(depends=["test_one"], name="test_two")
    def test_two(self):
        assert self.shared_data['value'] == 'data from test_one'
        # Update the variable
        TestSequence.shared_data['value'] = 'data from test_two'

    @pytest.mark.dependency(depends=["test_two"])
    def test_three(self):
        assert self.shared_data['value'] == 'data from test_two'
```

If a test fails, dependent tests will be skipped.

**Notice**: `pytest-dependency` is not being actively maintained.
